controller/scheduler-v2/scheduler.go

```diff
@@ -736,26 +763,51 @@ func (s *Scheduler) updateFormation(f *ct.Formation) (*Formation, error) {
 }
 
 func (s *Scheduler) startJob(req *JobRequest) (err error) {
-	log := logger.New("fn", "startJob", "job.type", req.Type, "job.restarts", req.restarts)
-	log.Info("starting job")
+	log := fnLogger("job.type", req.Type, "job.id", req.JobID)
+	log.Info("starting job", "job.restarts", req.restarts, "request.attempts", req.attempts)
+	s.pendingStarts.RemoveJob(req.Job)
+	// Copy on Write
+	job := *(req.Job)
+	newReq := &JobRequest{
+		RequestType: req.RequestType,
+		Job:         &job,
+	}
 	defer func() {
+		s.pendingStarts.AddJob(newReq.Job)
 		if err != nil {
-			s.scheduleJobRequest(req)
 			log.Error("error starting job", "err", err)
+			if newReq.attempts < maxJobAttempts {
```

Comment:
Limit number of times a failed job request can try to start. Rely on `Rectify` to handle the rest.